### PR TITLE
bazel execute request fields passed into scoot jobs through runner

### PR DESCRIPTION
**Pull request by dgdonut**
* ExecuteRequest data carried (optionally) in full down through to the Worker as part of runner.Command
* Adds data to internal thrifts: sched API (sagas) and worker API (running commands). Required some minor rewiring of thrift defs due to some thrift limitations. Normalized thrift docs around generation.
* ExecuteRequests now populate (enough) real fields in scheduled scoot jobs, although full integration w/ worker and get status is coming later
* Some minor refactoring

### CODE REVIEW

File: bazel/execution/request/definitions.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+import (
+	remoteexecution "google.golang.org/genproto/googleapis/devtools/remoteexecution/v1test"
+
+	requestthrift "github.com/twitter/scoot/bazel/execution/request/gen-go/request"
+	scootproto "github.com/twitter/scoot/common/proto"
+)
+
+// This type gives us a single reference point for passing Execute Requests and
+// leaves room for internal modifications if required
```
Comment (ryancouto) [inline bazel/execution/request/definitions.go:14]: Are there any particular internal modifications you think are likely to be necessary?

File: bazel/execution/request/request.thrift [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Structures for storing Bazel request fields in
+# tasks started from Bazel Execute Requests
+
+# See github.com/twitter/scoot/README.md for local Thrift prerequisites
```
Comment (ryancouto) [inline bazel/execution/request/request.thrift:4]: Somewhat obvious, but could you link e.g. https://godoc.org/google.golang.org/genproto/googleapis/devtools/remoteexecution/v1test#Digest next to struct declarations to clarify their origin

File: bazel/execution/service.go [hunk 3]
```diff
@@ -89,7 +91,7 @@ func (s *executionServer) Execute(
 		log.Errorf("Failed to schedule Scoot job: %s", err)
 		return nil, err
 	}
-	log.Info("Scheduled execute request as Scoot job: %s", id)
+	log.Infof("Scheduled execute request as Scoot job: %s", id)
```
Comment (ryancouto) [inline bazel/execution/service.go]: Could you structure this log to make it more easily searchable?

File: bazel/execution/service.go [hunk 4]
```diff
@@ -127,22 +129,30 @@ func (s *executionServer) Execute(
...
+	if err != nil {
+		log.Errorf("Failed to parse Timeout from Action: %s", err)
+		return result, err
+	}
+
 	var task sched.TaskDefinition
-	task.TaskID = fmt.Sprintf("ExecuteRequest_%d", time.Now().Unix())
-	task.Command.Argv = []string{"sleep", "10"}
+	task.TaskID = fmt.Sprintf("Bazel_ExecuteRequest_%s_%d", actionSha, time.Now().Unix())
+	task.Command.Argv = []string{"sleep", "10"} // used as a safe placeholder - worker will retrieve Command
```
Comment (ryancouto) [inline bazel/execution/service.go]: "worker will retrieve Command"
How? From where?

**Review by ryancouto — APPROVED** [review]

File: bazel/execution/request/definitions.go (hunk 1 above)
Comment (dgdonut) [reply bazel/execution/request/definitions.go:14]: Not really, if anything it gives us a type alias and potentially protects from changes in googleapis struct

File: bazel/execution/request/request.thrift (hunk 2 above)
Comment (dgdonut) [reply bazel/execution/request/request.thrift:4]: Sure will add

File: bazel/execution/service.go (hunk 3 above)
Comment (dgdonut) [reply bazel/execution/service.go]: Sure

File: bazel/execution/service.go (hunk 4 above)
Comment (dgdonut) [reply bazel/execution/service.go]: Worker will retrieve the command from the CAS, will make comment more clear

File: bazel/execution/request/definitions.go (hunk 1 above)
Comment (ryancouto) [reply bazel/execution/request/definitions.go:14]: I wonder if those are worth the slight complexity add. I don't have a strong opinion either way, up to you.

**Pull request merged**